lib/pbench/server/api/resources/database.py

```diff
@@ -0,0 +1,49 @@
+import sys
+from sqlalchemy import create_engine
+from sqlalchemy.orm import scoped_session, sessionmaker
+from sqlalchemy.ext.declarative import declarative_base
+from pbench.server import NoOptionError, NoSectionError
+
+
+class Database:
+    # Create declarative base model that our model can inherit from
+    Base = declarative_base()
+    # Initialize the db scoped session
+    db_session = scoped_session(sessionmaker(autocommit=False, autoflush=False))
+
+    @staticmethod
+    def get_engine_uri(config, logger):
+        try:
+            psql = config.get("Postgres", "db_uri")
+            return psql
+        except (NoSectionError, NoOptionError):
+            logger.error(
+                "Failed to find an [Postgres] section" " in configuration file."
+            )
+            sys.exit(1)
+
+        # return f"postgresql+{psql_driver}://{psql_username}:{psql_password}@{psql_host}:{psql_port}/{psql_db}"
+
+    @staticmethod
+    def init_engine(server_config, logger):
+        try:
+            return create_engine(Database.get_engine_uri(server_config, logger))
+        except Exception:
+            logger.exception("Exception while creating a sqlalchemy engine")
+            sys.exit(1)
+
+    @staticmethod
+    def init_db(server_config, logger):
+        # Make sure all the models are imported before this function gets called
```

Comment:
How does one ensure the models are imported before this function is called?  Is there a way for the code to detect that condition?

Reply:
Yeah I was doing it by manually importing the models here just to be safe `import pbench.server.api.resources.models` but that was resulting in a flake8 warning since I am not directly using the imports in code. Therefore I tried to investigate whether we really need to explicitly do the import statement here or can we get rid of it. I did that by putting `print(sys.modules.keys())`  right before we call `init_db` in the `lib/pbench/server/api/__init__.py` file and ran unit test file `test_user_auth.py` and it turns out the models does get imported before we reach to call the `init_db`. I am gussing this is probably because we import `pbench_users.py` in our `__init__.py` and `pbench_users` file imports the model it requires. 

Ideally I would put the `import pbench.server.api.resources.models` here just to be safe but it wont pass the flake8 test, but from what I see from running `print(sys.modules.keys())` before right before a call to `init_db` we dont need it actually.